src/components/tx/SignOrExecuteForm/SignForm.tsx

```diff
@@ -64,6 +65,8 @@ const SignForm = ({
 
   return (
     <form onSubmit={handleSubmit}>
+      {hasSigned && <ErrorMessage level="warning">You have already signed this transaction.</ErrorMessage>}
```

Comment:
Should we not move it into the `CheckWallet` component and add a new message there?

Reply:
CheckWallet doesn't know anything about transactions, it only checks the wallet itself.